aristanetworks/ansible-avd
# Refactor(eos_cli_config_gen)!: Require `enabled: true` under `vlan_interfaces.[].ip_attached_host_route_export`

## Change Summary


Require `enabled: true` under `vlan_interfaces.[].ip_attached_host_route_export`

## Related Issue(s)

Part of #2288 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes



To avoid ambiguous YAML data input, the data model for `vlan_interfaces.[].ip_attached_host_route_export` has been updated to require
an `enabled: true` key to be added.

Example with old data model (shown with old dictionary form input):

```yaml
vlan_interfaces:
  Vlan86:
    ip_address: 10.10.83.1/24
    ip_attached_host_route_export: {}
```

Same example with new data model (shown with new list form input):

```yaml
vlan_interfaces:
  - name: Vlan86
    ip_address: 10.10.83.1/24
    ip_attached_host_route_export:
      enabled: true
```

The change has been incorporated into `eos_designs` so action is only required when defining `structured_configuration` directly.

## How to test



- Updated molecule for eos_cli_config_gen. No config change.
- Reran eos_designs after updating logic. No config change. Expected change in structured.

## Checklist

### User Checklist


- N/A

### Repository Checklist
